src/gui/guiFormSpecMenu.cpp

```diff
@@ -2987,59 +2994,42 @@ void GUIFormSpecMenu::regenerateGui(v2u32 screensize)
 void GUIFormSpecMenu::legacySortElements(core::list<IGUIElement *>::Iterator from)
 {
 	/*
-	 * The order was:
-	 * (0. background colors and backgrounds)
-	 * 1. boxes
-	 * 2. all normal elements (like buttons)
-	 * 3. images
-	 * 4. item images
-	 * 5. item lists (TODO)
-	 */
+		Draw order for formspec_version <= 2:
+		-3  bgcolor
+		-2  background
+		-1  box
+		0   All other elements
+		1   image
+		2   item_image, item_image_button
+		3   list
+		4   label
+	*/
 
 	if (from == Children.end())
 		from = Children.begin();
 	else
 		from++;
 
 	core::list<IGUIElement *>::Iterator to = Children.end();
-
-	// step 1: move all boxes to back and all images to front
-	// (drawn in back = drawn first = in front of list)
-	for (auto i = from; i != to;) {
-		if (getTypeByID((*i)->getID()) == f_Box && i != from) {
-			auto e = *i;
-			i = Children.erase(i);
-			Children.insert_before(from, e);
-		} else if ((*i)->hasType(gui::EGUIET_IMAGE)) {
-			auto e = *i;
-			i = Children.erase(i);
-			Children.push_back(e);
-			if (to == Children.end())
-				to = Children.getLast(); // do not touch sorted back
-			if (i == Children.end())
-				break;
-		} else {
-			i++;
-		}
-	}
-
-	// step 2: move all item images to front
-	for (auto i = from; i != to;) {
-		if (getTypeByID((*i)->getID()) == f_ItemImage) {
-			auto e = *i;
-			i = Children.erase(i);
-			Children.push_back(e);
-			if (to == Children.end())
-				to = Children.getLast(); // do not touch sorted back
-			if (i == Children.end())
-				break;
-		} else {
-			i++;
-		}
+	// 1: Copy into a sortable container
+	std::vector<IGUIElement *> elements;
+	for (auto it = from; it != to; ++it)
+		elements.emplace_back(*it);
+
+	// 2: Sort the container
+	std::sort(elements.begin(), elements.end(),
+			[=] (const IGUIElement *a, const IGUIElement *b) -> bool {
```

Comment:
Is the capture-default needed?
```suggestion
			[] (const IGUIElement *a, const IGUIElement *b) -> bool {
```

Reply:
Yes. `[]` would isolate it entirely from the `guiFormSpec` class, meaning it has no access to `getSpecByID`.